plugins/convertPathData.js

```diff
@@ -766,11 +766,18 @@ function filters(path, params, { maybeHasStrokeAndLinecap, hasMarkerMid }) {
         }
       }
 
-      // remove useless non-first path segments
-      if (params.removeUseless && !maybeHasStrokeAndLinecap) {
-        // l 0,0 / h 0 / v 0 / q 0,0 0,0 / t 0,0 / c 0,0 0,0 0,0 / s 0,0 0,0
+      // remove useless non-start/end path segments
+      var isStart = !path[index - 1] || path[index - 1].command === 'm';
+      var isEnd = !path[index + 1] || path[index + 1].command === 'm';
+      if (
+        params.removeUseless &&
+        !hasMarkerMid &&
+        (!maybeHasStrokeAndLinecap || !(isStart || isEnd))
+      ) {
```

Comment:
`hasRoundLinejoinAndLinecap` from https://github.com/svg/svgo/pull/1751 could also be included.

```suggestion
      if (
        params.removeUseless &&
        !hasMarkerMid &&
        (hasRoundLinejoinAndLinecap || !maybeHasStrokeAndLinecap || !(isStart || isEnd))
      ) {
```

Reply:
This doesn't make sense as these do fundamentally different things.
If it has a linecap, we can't remove segments off the end. We can only remove segments in between.
If it doesn't have a linecap, we can remove all useless segments.
What you're proposing is that if it has a round linejoin, we can change `M5 5h0` to nothing. That would cause incorrect behavior, because the original shows up as a dot.

*I would personally write this as `maybeHasStrokeAndLinecap ? !isStart && !isEnd : true` as I find that way easier to read and understand, but style is personal.